src/data/sym/card.lean

```diff
@@ -36,9 +35,108 @@ Prove the general case of stars and bars.
 stars and bars
 -/
 
+def multichoose1 (n k : ℕ) := fintype.card (sym (fin n) k)
+
+def multichoose2 (n k : ℕ) := (n + k - 1).choose k
+
+def encode (n k : ℕ) (x : sym (fin n.succ) k.succ) : sym (fin n) k.succ ⊕ sym (fin n.succ) k :=
+if h : fin.last n ∈ x then
+  sum.inr ⟨x.val.erase (fin.last n), by { rw [multiset.card_erase_of_mem h, x.property], refl }⟩
+else begin
+  refine sum.inl (x.map (λ a, ⟨if (a : ℕ) = n then 0 else a, _⟩)),
```

Comment:
```suggestion
  refine sum.inl (x.map fin.cast_pred),
```

Reply:
I don't think there is a clean way to use `fin.cast_pred`. `fin.cast_pred` takes a `fin (n + 2)` and returns a `fin (n + 1)`. Maybe I'm missing something.